### PR TITLE
Fix: Ability to provide newtype for sequence comprehension length

### LINKED ISSUE #5065: Sequence comprehension and Type descriptors issue
### Dafny version

latest-nightly

### Code to produce this issue

```dafny
const TWO_TO_THE_8: int := 0x100
newtype uint8 = x: int | 0 <= x < TWO_TO_THE_8

method Main() {
  var i: uint8 := 10;
  var s: seq<uint8> := seq(i, _ => 0);
  print s;
}
```

### Command to run and resulting output

```code
dafny run -t:java Example.dfy


error: method Create in class DafnySequence<T#2> cannot be applied to given types;
    _1_s = dafny.DafnySequence.Create(uint8._typeDescriptor(), _0_i, ((java.util.function.Function<java.math.BigInteger, java.lang.Byte>)(_2___v0_boxed0) -> {
```

### What happened?

Compilation fails

Related: https://github.com/dafny-lang/dafny/issues/3952

### What type of operating system are you experiencing the problem on?

Windows

### PR DESCRIPTION
**Pull request by MikaelMayer**
This PR fixes #5065
I added the corresponding test.

<small>By submitting this pull request, I confirm that my contribution is made under the terms of the [MIT license](https://github.com/dafny-lang/dafny/blob/master/LICENSE.txt).</small>

### CODE REVIEW

File: Source/DafnyCore/Compilers/Dafny/Compiler-dafny.cs [hunk 1]
```diff
@@ -1796,7 +1796,11 @@ private class ExprLvalue : ILvalue {
         EmitExpr(expr.Initializer, inLetExprBody, new BuilderSyntaxTree<ExprContainer>(elem), wStmts);
 
         var n = new ExprBuffer(null);
-        EmitExpr(expr.N, inLetExprBody, new BuilderSyntaxTree<ExprContainer>(n), wStmts);
+        var size = expr.N;
+        if (size.Type.AsNewtype is { }) {
```
Comment (MikaelMayer) [inline Source/DafnyCore/Compilers/Dafny/Compiler-dafny.cs:1800]: This is untested for now but will be soon as I'm writing the Rust compiler that leverages this experimental compiler.

File: docs/dev/news/5065.fix [hunk 2]
```diff
@@ -0,0 +1 @@
+Ability to provide newtype for sequence comprehension length
```
Comment (robin-aws) [inline docs/dev/news/5065.fix:1]:
```suggestion
Fixed support for newtypes as sequence comprehension lengths in Java
```

Just tweaking to be clear it's not a new feature, just a Java-specific fix.

File: Source/DafnyCore/Compilers/Dafny/Compiler-dafny.cs (hunk 1 above)
Comment (robin-aws) [reply Source/DafnyCore/Compilers/Dafny/Compiler-dafny.cs:1800]: I assume that means the test file doesn't work for Rust for an unrelated reason?

**Review by robin-aws — CHANGES_REQUESTED** [review]
You beat me to it. :) Just want to tweak the release note.

**Review by robin-aws — APPROVED** [review]

File: Source/DafnyCore/Compilers/Dafny/Compiler-dafny.cs (hunk 1 above)
Comment (MikaelMayer) [reply Source/DafnyCore/Compilers/Dafny/Compiler-dafny.cs:1800]: Indeed. This part is totally untested for now.

**Pull request merged**